### PR TITLE
Plug Dafny Backend into CI

### PR DESCRIPTION
**Pull request by jtristan**
Adding new Dafny backend for featherweight Dafny to plug into CI

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Comment by jtristan** [comment]
Before building, need to run Source/DafnyCore/DafnyGeneratedFromDafny.sh
You then need to remove the DAST namespace from either the FDafny or the Rust generated code

File: Source/DafnyCore/Compilers/ResolvedDesugaredExecutableDafny/Compiler-fdafny.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using System;
+using System.IO;
+using System.Linq;
+using Dafny;
+using FDafnyPlugin;
+
+namespace Microsoft.Dafny.Compilers;
+
+class RDEDafnyCompiler : DafnyWrittenCompiler {
```
Comment (MikaelMayer) [inline Source/DafnyCore/Compilers/ResolvedDesugaredExecutableDafny/Compiler-fdafny.cs:9]: Please either use full words or add a comment about what the abbreviation means.

**Review by MikaelMayer — COMMENTED** [review]
Requesting a few changes.

File: Source/DafnyCore/Compilers/ResolvedDesugaredExecutableDafny/ResolvedDesugaredExecutableDafnyBackend.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Dafny.Compilers;
+
+public class ResolvedDesugaredExecutableDafnyBackend : DafnyExecutableBackend {
+
+  public override IReadOnlySet<string> SupportedExtensions => new HashSet<string> { ".dfy" };
```
[the remark this answers is not in the document]
Comment (jtristan) [reply Source/DafnyCore/Compilers/ResolvedDesugaredExecutableDafny/ResolvedDesugaredExecutableDafnyBackend.cs:9]: We don't need to differentiate them, they are valid dafny programs.

**Review by MikaelMayer — APPROVED** [review]
Great job making sure this compiler is development is integrated in the CI.

**Pull request merged**